libtransmission/peer-mgr-wishlist.cc

```diff
@@ -14,164 +15,312 @@
 
 #include "libtransmission/transmission.h"
 
+#include "libtransmission/bitfield.h"
 #include "libtransmission/crypto-utils.h" // for tr_salt_shaker
 #include "libtransmission/peer-mgr-wishlist.h"
-#include "libtransmission/utils.h"
 
 namespace
 {
-using SaltType = tr_piece_index_t;
-
-struct Candidate
+std::vector<tr_block_span_t> make_spans(small::set<tr_block_index_t> const& blocks)
 {
-    tr_piece_index_t piece;
-    size_t n_blocks_missing;
-    tr_priority_t priority;
-    SaltType salt;
-
-    Candidate(tr_piece_index_t piece_in, size_t missing_in, tr_priority_t priority_in, SaltType salt_in)
-        : piece{ piece_in }
-        , n_blocks_missing{ missing_in }
-        , priority{ priority_in }
-        , salt{ salt_in }
+    if (std::empty(blocks))
     {
+        return {};
     }
 
-    [[nodiscard]] constexpr auto compare(Candidate const& that) const noexcept // <=>
+    auto spans = std::vector<tr_block_span_t>{};
```

Comment:
_(nit)_ we can prealloc with `spans.reserve(std::size(blocks));`

Reply:
5524f03